core/services/ccip/orm_test.go

```diff
@@ -0,0 +1,352 @@
+package ccip
+
+import (
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccip"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils/pgtest"
+)
+
+func setupORM(t *testing.T) (ORM, sqlutil.DataSource) {
+	t.Helper()
+
+	db := pgtest.NewSqlxDB(t)
+	orm, err := NewORM(db)
+
+	require.NoError(t, err)
+
+	return orm, db
+}
+
+func generateChainSelectors(n int) []uint64 {
+	selectors := make([]uint64, n)
+	for i := 0; i < n; i++ {
+		selectors[i] = rand.Uint64()
+	}
+
+	return selectors
+}
+
+func generateGasPriceUpdates(chainSelector uint64, n int) []GasPriceUpdate {
+	updates := make([]GasPriceUpdate, n)
+	for i := 0; i < n; i++ {
+		row := GasPriceUpdate{
+			SourceChainSelector: chainSelector,
+			GasPrice:            new(big.Int).Mul(big.NewInt(1e18), big.NewInt(int64(i))),
+		}
+		updates[i] = row
+	}
+
+	return updates
+}
+
+func generateTokenAddresses(n int) []ccip.Address {
+	addrs := make([]ccip.Address, n)
+	for i := 0; i < n; i++ {
+		addrs[i] = ccip.Address(utils.RandomAddress().Hex())
+	}
+
+	return addrs
+}
+
+func generateTokenPriceUpdates(tokenAddr ccip.Address, n int) []TokenPriceUpdate {
+	updates := make([]TokenPriceUpdate, n)
+	for i := 0; i < n; i++ {
+		row := TokenPriceUpdate{
+			TokenAddr:  tokenAddr,
+			TokenPrice: new(big.Int).Mul(big.NewInt(1e18), big.NewInt(int64(i))),
+		}
+		updates[i] = row
+	}
+
+	return updates
+}
+
+func getGasTableRowCount(t *testing.T, ds sqlutil.DataSource) int {
+	var count []int
```

Comment:
i believe if you declare this as `count int` then SelectContext will 'just do the right thing' and you won't have to index the slice in the return statemtn

Reply:
I tried that initially, but got `expected slice but got int`

Reply:
Maybe QueryRow?